sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -405,6 +418,154 @@ private object CosmosAccountConfig {
   }
 }
 
+object CosmosAuthType extends Enumeration {
+    type CosmosAuthType = Value
+    val MasterKey, ServicePrinciple = Value
+}
+
+private object AzureEnvironmentType extends Enumeration {
+    type AzureEnvironmentType = Value
+    val Azure, AzureChina, AzureUsGovernment, AzureGermany = Value
+}
+
+trait CosmosAuthConfig {}
+
+private case class CosmosMasterKeyAuthConfig(accountKey: String) extends CosmosAuthConfig
+private case class CosmosAadAuthConfig(subscriptionId: String,
```

Comment:
Should this be named CosmosAadServciePrincipleAadAuthConfig? Or would you expect it to be a super set across all add auth types we eventually support?

Reply:
it is the latter and also I was not sure what kind others we are going to add in the future, but open to change this class to be more specific to Service principle